pkg/controller/persistentvolume/types.go

```diff
@@ -105,19 +115,29 @@ func (pvIndex *persistentVolumeOrderedIndex) Find(searchPV *api.PersistentVolume
 		// return the exact pre-binding match, if found
 		unboundVolumes := []*api.PersistentVolume{}
 		for _, volume := range volumes {
```

Comment:
How about reorganizing this block.  Looks like the logic is 
```go
if volume has [createdForKey] and it doesn't match prebound claim, skip
if volume has createdForKey and it does match prebound claim, return it
if volume doesn't have a spec claim ref, add it to the list
if volume does have a spec claim ref that matches our preboundClaim, return it
```

Can you make that a switch case?